pkg/providers/instancetype/types.go

```diff
@@ -377,31 +376,25 @@ func kubeReservedResources(cpus, pods, eniLimitedPods *resource.Quantity, amiFam
 			resources[v1.ResourceCPU] = *cpuOverhead
 		}
 	}
-	if kc != nil && kc.KubeReserved != nil {
-		kubeReserved := lo.MapEntries(kc.KubeReserved, func(k string, v string) (v1.ResourceName, resource.Quantity) {
-			return v1.ResourceName(k), resource.MustParse(v)
-		})
-		return lo.Assign(resources, kubeReserved)
-	}
-	return resources
+	kubeReservedResourceList := lo.MapEntries(kubeReserved, func(k string, v string) (v1.ResourceName, resource.Quantity) {
```

Comment:
nit: Consider combining this line with L382 since you can start to chain the `lo.Assign` with the `lo.MapEntries` now